chain-vote/src/gang/babystep.rs

```diff
@@ -0,0 +1,110 @@
+use super::p256k1::*;
+use rayon::prelude::*;
+use std::collections::HashMap;
+use std::convert::TryFrom;
+
+// make steps asymmetric, in order to better use caching of baby steps.
+// balance of 2 means that baby steps are 2 time more than sqrt(MAX_VOTES_BSGS)
+const DEFAULT_BALANCE: u64 = 2;
+
+/// Holds precomputed baby steps for the baby-stap giant-step algorithm
+/// for solving discrete log on ECC and tally private votes
+pub struct PrivateTallyTable {
```

Comment:
Hmm, maybe the internal name in this module should not refer to a private tally, because the rest of `gang` is just group theory math.